pandas/_libs/src/ujson/python/objToJSON.c

```diff
@@ -571,7 +537,6 @@ static int NpyTypeToJSONType(PyObject *obj, JSONTypeContext *tc, int npyType,
             return JT_NULL;
         }
         GET_TC(tc)->doubleValue = (double)doubleVal;
-        GET_TC(tc)->PyTypeToJSON = CDouble;
```

Comment:
To make sure i understatnd, Foo._PyTypeToJSON are the callbacks you're referring to?

Reply:
Yep that's right - anything getting assigned to `PyTypeToJSON` (side note - not technically callbacks but void pointer functions; edited appropriately)

In the end we might be able to get rid of the custom `PyEncoder` altogether and just use more built-in ujson functionality by assigning the native types and returning `JT_<TYPE>` where appropriate